src/main.ts

```diff
@@ -0,0 +1,248 @@
+import * as fs from "fs";
+import * as path from "path";
+import axios, {isAxiosError} from 'axios';
+
+import * as core from "@actions/core";
+
+import * as common from "./common";
+
+/** file creation info */
+interface FileInfo {
+    /** file name */
+    name: string;
+    /** file contents */
+    contents: string;
+    /** creation options */
+    options: fs.WriteFileOptions;
+    /** file must not exist when creating */
+    mustNotExist: boolean;
+}
+
+/** default known_hosts */
+const KNOWN_HOSTS = [
+    "github.com ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCj7ndNxQowgcQnjshcLrqPEiiphnt+VTTvDP6mHBL9j1aNUkY4Ue1gvwnGLVlOhGeYrnZaMgRK6+PKCUXaDbC7qtbW8gIkhL7aGCsOr/C56SJMy/BCZfxd1nWzAOxSDPgVsmerOBYfNqltV9/hWCqBywINIR+5dIg6JTJ72pcEpEjcYgXkE2YEFXV1JHnsKgbLWNlhScqb2UmyRkQyytRLtL+38TGxkxCflmO+5Z8CSSNY7GidjMIZ7Q4zMjA2n1nGrlTDkzwDCsw+wqFPGQA179cnfGWOWRVruj16z6XyvxvjJwbz0wQZ75XK5tKSb7FNyeIEs4TT4jk+S4dhPeAUC5y+bDYirYgM4GC7uEnztnZyaVWQ7B381AK4Qdrwt51ZqExKbQpTUNn+EjqoTwvqNj4kqx5QUCI0ThS/YkOxJCXmPUWZbhjpCg56i+2aB6CmK2JGhn57K5mj0MNdBXA4/WnwH6XoPWJzK5Nyu2zB3nAZp+S5hpQs+p1vN1/wsjk=",
```

Comment:
do you know what this known hosts is?

Reply:
know hosts here represent list of servers with encryptions type & public key (contains the list of server that can be relied on not an imposter). I don't actually know the context behind the host provided here. Here is a small description about it by the author https://github.com/shimataro/ssh-key-action?tab=readme-ov-file#i-want-to-omit-known_hosts

Reply:
we can refer this in the code as a comment: https://github.blog/news-insights/company-news/we-updated-our-rsa-ssh-host-key/